portfolio/src/main/java/com/google/sps/servlets/UpvoteCommentServlet.java

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.util.List;
+import java.util.ArrayList;
+import com.google.gson.Gson;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+
+
+import com.google.sps.data.Comment;
+
+/** Servlet responsible for deleting comments. */
+@WebServlet("/upvote-comment")
+public class UpvoteCommentServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long id = Long.parseLong(request.getParameter("id"));
+    long upvotes = Long.parseLong(request.getParameter("upvotes"));
+    long timestamp = Long.parseLong(request.getParameter("timestamp"));
+    String comment = request.getParameter("comment");
+
+    Key taskEntityKey = KeyFactory.createKey("Comment", id);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.delete(taskEntityKey);
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("comment", comment);
+    commentEntity.setProperty("timestamp", timestamp);
+    commentEntity.setProperty("upvotes", upvotes);
+    commentEntity.setProperty("id", id);
+
```

Comment:
This looks very different from the upVote logic we worked on in the previous PR. Confirm that you have merged correctly with master.

Reply:
Oops! Was waiting to merge upvote feature and then do the resolve merge conflict on this file; forgot to do that yesterday afternoon. Yes, merged correctly, this just needs to be caught up. Doing now.

Reply:
Finished merging and such; I also updated the upvote logic per your last comment. It should now handle simultaneous upvoting.